engine/samples/imgui/main.cpp

```diff
@@ -1,12 +1,92 @@
-/// [Full code]
 /// [Including plugin headers]
+#include <iostream>
+#include <map>
+#include <vector>
+
 #include <imgui.h>
 
+#include <cubos/core/reflection/external/map.hpp>
+#include <cubos/core/reflection/external/primitives.hpp>
+#include <cubos/core/reflection/external/string.hpp>
+#include <cubos/core/reflection/external/vector.hpp>
+#include <cubos/core/reflection/reflect.hpp>
+#include <cubos/core/reflection/traits/fields.hpp>
+
+#include <cubos/engine/imgui/data_inspector.hpp>
 #include <cubos/engine/imgui/plugin.hpp>
+
 /// [Including plugin headers]
 
+using cubos::core::ecs::Read;
+using cubos::core::ecs::Write;
+using cubos::core::reflection::reflect;
+using cubos::engine::DataInspector;
 using namespace cubos::engine;
 
+/// [Creating a dummy resource]
+struct Person
+{
+    CUBOS_REFLECT;
+    std::string name;
+    int32_t age;
+    float weight;
+    bool dead;
+};
+
+CUBOS_REFLECT_IMPL(Person)
+{
+    return cubos::core::reflection::Type::create("Person").with(cubos::core::reflection::FieldsTrait()
```

Comment:
```suggestion
    return Type::create("Person").with(FieldsTrait()
```

Add using statements for this